tasks.py

```diff
@@ -239,13 +240,14 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None, protocol=None, nod
     run("kubectl delete po -nmetallb-system --all", echo=True)
 
     manifests_dir = os.getcwd() + "/manifests"
+    curr_branch = subprocess.check_output("git rev-parse --abbrev-ref HEAD", shell=True).decode().strip("\n")
     with tempfile.TemporaryDirectory() as tmpdir:
         # Copy namespace manifest.
         shutil.copy(manifests_dir + "/namespace.yaml", tmpdir)
 
         with open(manifests_dir + "/metallb.yaml") as f:
             manifest = f.read()
-        manifest = manifest.replace(":main", ":dev-{}".format(architecture))
+        manifest = re.sub(":{}\.?[0-9]?".format(curr_branch), ":dev-{}".format(architecture), manifest)
```

Comment:
Does this still work if what it needs to replace is `main` ?  Also, this looks a bit risky.  There are a *lot* of lines that include `:` in a yaml doc.  Can you make sure we're only matching the `image:` lines?

Reply:
I

> Does this still work if what it needs to replace is `main` ? Also, this looks a bit risky. There are a _lot_ of lines that include `:` in a yaml doc. Can you make sure we're only matching the `image:` lines?

yes I tried both v0.10 and main as showing the UT description, this is matching for :branch_name so not sure where is the risk ?

Reply:
sorry, i didn't read it correctly